src/anomalib/models/components/base/anomaly_module.py

```diff
@@ -234,11 +235,20 @@ def _load_normalization_class(self, state_dict: OrderedDict[str, Tensor]) -> Non
         else:
             warn("No known normalization found in model weights.")
 
+    def _load_pixel_metrics(self, state_dict: OrderedDict[str, Tensor]) -> None:
+        """Create pixel_metrics and add AUPRO."""
+        if not hasattr(self, "pixel_metrics") and "pixel_metrics.AUPRO.fpr_limit" in state_dict.keys():
+            self.pixel_metrics = AnomalibMetricCollection([], prefix="pixel_")
+            fpr_limit = state_dict["pixel_metrics.AUPRO.fpr_limit"].item()
+            self.pixel_metrics.add_metrics(AUPRO(fpr_limit=fpr_limit))
```

Comment:
I am in favour of something more generic and less hard-coded. Maybe something like
```python
def _load_metrics(self, state_dict: OrderedDict[str, Tensor]) -> None:
        """Create pixel_metrics and add AUPRO."""
        self._set_metrics("pixel", state_dict)
        self._set_metrics("image", state_dict)

    def _set_metrics(self, name: str, state_dict: OrderedDict[str, Tensor]):
        """Sets the pixel/image metrics.

        Args:
        name (str): is it pixel or image.
        state_dict (OrderedDict[str, Tensor]): state dict of the model.
        """
        metric_keys = [key for key in state_dict.keys() if key.startswith(f"{name}_metrics")]
        if not hasattr(self, f"{name}_metrics") and any(metric_keys):
            metrics = AnomalibMetricCollection([], prefix=f"{name}_")
            for key in metric_keys:
                class_name = key.split(".")[1]
                try:
                    metrics_module = importlib.import_module("anomalib.utils.metrics")
                    metrics_cls = getattr(metrics_module, class_name)
                except Exception as exception:
                    raise ImportError(f"Class {class_name} not found in module anomalib.utils.metrics") from exception
                metrics.add_metrics(metrics_cls())
            setattr(self, f"{name}_metrics", metrics)
```